### ISSUE TITLE
Fold Options into Preferences.

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/joist#837]
Related to #835, and summary of Zoom discussion with @jessegreenberg ...

* Any sim that has an Options dialog (uses SimOptions `createOptionsDialogContent`) will need to be folded into Preferences.  
* This will probably involve using SimOptions `preferencesConfiguration` instead of `createOptionsDialogContent`. 
* If the Options dialog uses ProjectorModeCheckbox, it should be removed, and instead added to the Preferences "Visual" tab. 

These sims use `createOptionsDialogContent` but only have a ProjectorModeCheckbox:
- [x] atomic-interactions
- [x] blackbody-spectrum
- [x] charges-and-fields
- [x] coulombs-law
- [x] diffusion
- [x] models-of-the-hydrogen-atom
- [x] rutherford-scattering
- [x] states-of-matter
- [x] states-of-matter-basics

These sims use `createOptionsDialogContent` and have controls that should appear in Preferences "Visual" tab:
- [x] build-an-atom
- [x] center-and-variability
- [x] circuit-construction-kit-common
- [x] gas-properties
- [x] gases-intro
- [x] geometric-optics
- [x] molecule-polarity
- [x] molecule-shapes
- [x] molecule-shapes-basics
- [x] tambo

**Comment by zepumph** [comment]
> These sims use `createOptionsDialogContent` and have controls that should appear in Preferences "Visual" tab:

In that case, it will probably want to hold off until #835 has been worked out. If options are totally sim-specific, then we can proceed using `PreferencesConfiguration.createSimControls`

**Comment by zepumph** [comment]
Definitely on hold until we finish https://github.com/phetsims/joist/issues/835, there is a new API for creating custom preferences, even in the general tab.

**Comment by zepumph** [comment]
Now unblocked! Yay

**Comment by zepumph** [comment]
I converted the ProjectorMode-checkbox-only sims over to preferences above. It lost PhET-iO instrumentation on some, but consistency is good. If we want the checkbox instrumented in the future, we will add it then.

**Comment by zepumph** [comment]
The second half of the list has been done above. A couple more things to do:

- [ ] `new ProjectorModeCheckbox` should not be used, we should be moving towards the preferencesModel setting.
- [ ] OptionsDialog should be deleted. Right now the main challenge is that OptionsDialog has constants used elsewhere.
- [ ] GreenhouseEffectQueryParameters.optionsDialog should probably be renamed
- [ ] *OptionsDialogContent should probably be renamed to *PreferencesContent


That's all I have time for tonght!

**Comment by jessegreenberg** [comment]
Excellent! I am seeing a couple of phet-io problems on CT in diffucion/gas-properties that I don't know how to fix 
> Assertion failed:  phetioID already registered: diffusion.general.view.navigationBar.preferencesButton.preferencesDialogCapsule.preferencesDialog.preferencesPanels.generalPreferencesPane

**Comment by pixelzoom** [comment]
A couple of other things that need to be done:

- [ ] Rename "Options" classes/variables/methods/etc. like GOOptionsNode.js. Should we use "Preferences", e.g. GOPreferencesNode?

- [ ] Update documentation like this that appears in GOOptionsNode.ts:
> // GOOptionsNode is the user interface for global options, accessed via PhET > Options.

**Comment by zepumph** [comment]
>Excellent! I am seeing a couple of phet-io problems on CT in diffucion/gas-properties/geometric-optics that I don't know how to fix

Fixed above. We changed the tandem API so now you are given a parent tandem, instead of the tandem you should use.

**Comment by pixelzoom** [comment]
- [ ] GasPropertiesGlobalOptionsNode is still creating a ProjectorModeCheckbox.

**Comment by zepumph** [comment]
See above: >new ProjectorModeCheckbox should not be used, we should be moving towards the preferencesModel setting

**Comment by pixelzoom** [comment]
I see, I misunderstood.  I thought that the point of addressing the first checklist in https://github.com/phetsims/joist/issues/837#issue-1346723654 was to delete ProjectorModeCheckBox and use `supportsProjectorMode: true`.

**Comment by zepumph** [comment]
Righto! 

I deleted ProjectorModeCheckbox above. The only weird thing is that diffusion still creates a GasProperties preferences Node even though it has no content for it. Since it is in a suite. I thought that was better, in case a new option is added to that sime that should apply to all three sims (gas-properties, diffusion, and gases-intro).

Also, new tandem names across the board will fix CT. I went with `simPreferencesContent` generally.

**Comment by zepumph** [comment]
@jessegreenberg, over to you for next steps!

**Comment by pixelzoom** [comment]
> I deleted ProjectorModeCheckbox above. The only weird thing is that diffusion still creates a GasProperties preferences Node even though it has no content for it. Since it is in a suite. I thought that was better, in case a new option is added to that sime that should apply to all three sims (gas-properties, diffusion, and gases-intro).

Not good. To be addressed in https://github.com/phetsims/diffusion/issues/8.

**Comment by pixelzoom** [comment]
> Also, new tandem names across the board will fix CT. I went with `simPreferencesContent` generally.

There's nothing in that name that indicates that it's content for the General tab. What if the sim is also adding sim-specific options to other tabs? (That's certainly supported by PreferencesModel.).  How about `{{SIM}}{{TAB}}PreferencesContent` as a convention.  For example, `geometricOpticsGeneralPreferencesContent` for _General_ tab content added by Geometric Options.

**Comment by pixelzoom** [comment]
> If a sim adds a visual customPreference and a generalCustomPreference, shouldn't it be named something better than `simVisualPreference` and `simGeneralPreference`? Why not name to what the content actually has?

Sure, that would be fine.  But that's not what you proposed above. You proposed convention `simPreferencesContent`.  And I'm pointing out the shortcomings of that convention - it assumes that you're only adding custom content to the _General_, e.g.:

```js
preferencesModel: new PreferencesModel( {
  generalOptions: {
    customPreferences: [ {
      createContent: ...  // tandem name = 'simPreferencesContent'
    } ]
  }
} )
```

The Preferences API supports adding custom content to any tab.  So if we go with your simPreferencesContent` convention for the _General tab_... What do you propose to use for the tandem names for the custom content in the Visual and Audio tabs in this example?

```js
preferencesModel: new PreferencesModel( {
  generalOptions: {
    customPreferences: [ {
      createContent: ...  // tandem name = 'simPreferencesContent'
    } ]
  },
  visualOptions: {
    customPreferences: [ {
      createContent: ...  // tandem name = ???
    } ]
  },
  audiolOptions: {
    customPreferences: [ {
      createContent: ...  // tandem name = ???
    } ]
  }
} )
```

**Comment by pixelzoom** [comment]
> If a sim adds a visual customPreference and a generalCustomPreference, shouldn't it be named something better than simVisualPreference and simGeneralPreference? Why not name to what the content actually has?

A reason not to "name to what the content actually has" is PhET-iO standardization.  I can imagine that PhET-iO designers will want to have consistent tandem names for "custom content added to the General tab", "custom content added to the Audio tab", etc.

**Comment by pixelzoom** [comment]
I discussed the tandem name issue with @zepumph.   We agreed that `simPreferences` is a better general name, for use in all tabs.  The phetioID (tandem hierarchy) will tell you which tab you're in. 

For example, instead of this, where "geometricOpticsPreferencesContent" is a bt redundant:

`geometricOptics.general.view.navigationBar.preferencesButton.preferencesDialogCapsule.archetype.preferencesPanels.generalPreferencesPanel.geometricOpticsPreferencesContent`

... we'll have this:

`geometricOptics.general.view.navigationBar.preferencesButton.preferencesDialogCapsule.archetype.preferencesPanels.generalPreferencesPanel.simPreferences`

**Comment by pixelzoom** [comment]
- [ ] GOOptions.ts and `geometricOptics.global.view.options` need to be renamed.  Should they be renamed to GOPreferences.ts and `geometricOptics.global.view.preferences`?   Note that `geometricOptics.global.view.options.cueingArrowsEnabledProperty` will NOT appear in the Preferences dialog, because (by design) it has no presence in the UI and is a PhET-iO-only Property.

**Comment by pixelzoom** [comment]
Things that need to be renamed in my sims:

- [ ] GOOptionsNode, GOOptions, associated tandems
- [ ] GasPropertiesGlobalOptionsNode, GasPropertiesGlobalOptions, associated tandems
- [ ] MPOptionsNode, MPGlobalOptions, associated tandems

It looks like other sims have similar renaming issues (e.g. build-an-atom GlobalOptionsNode) but I'm not going to attempt to enumerate them.

**Comment by jessegreenberg** [comment]
I think we are done here except for the items listed in https://github.com/phetsims/joist/issues/837#issuecomment-1224750873 and https://github.com/phetsims/joist/issues/837#issuecomment-1224787364. @pixelzoom can you please assign this back to me when those are done? I reviewed the lists in the original issue and sims are looking OK. I looked for remaining references to 'OptionsDialog' or `GlobalOptions` or 'options content` and other combinations in class names and documentation and updated them. I noticed in https://github.com/phetsims/molecule-shapes-basics/commit/39077fd3b03aa6e9e797fe9fd78031a37f51e892 we lost projector mode in molecule-shapes-basics. Id like to see if QA would be able to review and look for layout problems, missing or broken preferences, or anything else before closing.

**Comment by pixelzoom** [comment]
I complete geometric-optics, molecule-polarity, and gas-properties in their respective issues (linked above).

General conventions I used:

* Preferences Properties are under `{sim}.global.model.preferences` in PhET-iO

* tandem name for the Node returned by SimOptions.preferencesModel.generalOptions.customPreferences.createContent is 'simPreferences', as decided in https://github.com/phetsims/joist/issues/837#issuecomment-1224743329

**Comment by pixelzoom** [comment]
Looks like there's still work to be done here. We decided to use 'simPreferences' as the standard tandem name.  That's being violated in every sim that I've looked at.  Examples...

build-an-atom-main.js:

```js
        createContent: tandem => new VisualPreferencesNode( BAAGlobalPreferences.highContrastParticlesProperty,
          tandem.createTandem( 'buildAnAtomPreferencesContent' ) )
```

cck-ac-virtual-lab-main.js

```js
        createContent: tandem => new CCKCGeneralPreferencesContentNode( tandem.createTandem( 'cckPreferencesContent' ) )
```

cck-virtual-lab-main.js:

```js
createContent: tandem => new CCKCGeneralPreferencesContentNode( tandem.createTandem( 'cckPreferencesContent' ) )
```

cck-dc-main.js:

```js
createContent: tandem => new CCKCGeneralPreferencesContentNode( tandem.createTandem( 'cckPreferencesContent' ) )
```

energy-skate-park-main.js:

```js
createContent: tandem => new EnergySkateParkPreferencesNode( energySkateParkPreferencesModel, tandem.createTandem( 'energySkateParkPreferencesContent' ) )
```

Back to @jessegreenberg.

**Comment by pixelzoom** [comment]
There's also no standard for preferences Properties.  In all of my sims, they are under `{sim}.global.model.preferences`. 

I see `energySkatePark.global.model.energySkateParkPreferencesModel`, which is redunant in 2 ways -- "energySkatePark" and "model", both of which are already in the phetioID.     I haven't looked at other sims.

Recommended to choose a convention here, and use it consistently. I'm partial to `{sim}.global.model.preferences`.

**Comment by pixelzoom** [comment]
On Slack, we discussed the `{sim}.global.model.preferences` convention.  

Instead of defining this everywhere, like I've done in my sims:

`const preferencesTandem = Tandem.GLOBAL_MODEL.createTandem( 'preferences' );`

I've added this to Tandem.ts:

```
/**
 * Use this as the parent tandem for Properties that are related to sim-specific preferences.
 */
Tandem.PREFERENCES = Tandem.GLOBAL_MODEL.createTandem( 'preferences' );
```

So we should use `Tandem.PREFERENCES` for preferences Properties. And if the designers decide they want preferences Properties somewhere else, we can just change `Tandem.PREFERENCES`.

**Comment by jessegreenberg** [comment]
OK, the PhET-iO changes described in the previous three comments are done.  I think everything discussed in this issue is now complete. @zepumph can you think of anything else to review or do here?

**Comment by zepumph** [comment]
All looks good here. Thanks @pixelzoom and @jessegreenberg!!! Ready to close?

**Comment by zepumph** [comment]
Please note that we will be adding a PhET-iO migration rule because we removed the OptionsDialog menu item (and whole dialog). See https://github.com/phetsims/gravity-and-orbits/issues/434

**Comment by jessegreenberg** [comment]
Commits above look good. I believe so, closing.
